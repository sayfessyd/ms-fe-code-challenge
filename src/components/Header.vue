<template>
  <section :class="'hero is-small is-bold '+$globals.class_hero">
    <div class="hero-body">
      <div class="container has-text-centered">
        <img src="../assets/Twitter_bird_icon.png" width="100px" alt="Twitter Logo">
        <h1 class="title">{{title}}</h1>
        <h2 class="subtitle">{{message}}</h2>
      </div>
    </div>
    <!-- Hero footer: will stick at the bottom -->
    <div class="hero-foot">
      <nav class="tabs is-boxed is-fullwidth">
        <div class="container has-text-centered">
          <ul class="is-responsive">
            <li class="is-active">
              <div class="control has-icons-left">
                <div class="select is-small is-rounded">
                  <select v-model="$globals.order">
                    <option v-for="order of orders">{{order}}</option>
                  </select>
                </div>
                <span class="icon is-small is-left">
                  <i class="fas fa-sort-alpha-up"></i>
                </span>
              </div>
            </li>
            <li>
              <div class="control has-icons-left">
                <div class="select is-small is-rounded">
                  <select v-model="$globals.time_range">
                    <option v-for="time_range of time_ranges">{{time_range}}</option>
                  </select>
                </div>
                <span class="icon is-small is-left">
                  <i class="fas fa-clock"></i>
                </span>
              </div>
            </li>
            <li>
              <div class="control has-icons-left">
                <div class="select is-small is-rounded">
                  <select v-model="$globals.tweet_number">
                    <option v-for="tweet_number of tweet_numbers">{{tweet_number}}</option>
                  </select>
                </div>
                <span class="icon is-small is-left">
                  <i class="fas fa-file"></i>
                </span>
              </div>
            </li>
            <li>
              <div class="control has-icons-left">
                <div class="select is-small is-rounded">
                  <select v-model="$globals.pallete">
                    <option v-for="pallete of palletes">{{pallete}}</option>
                  </select>
                </div>
                <span class="icon is-small is-left">
                  <i class="fas fa-palette"></i>
                </span>
              </div>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </section>
</template>

<script>
  // Import the globalState variable
  import { globalState } from "../main.js";

  export default {
    name: "Header",
    data() {
      return {
        message:
          "Last 30 tweets from @MakeSchool, @newsycombinator and @ycombinator",
        title: "Twitter API",
        palletes: ["Primary", "Info", "Light", "Dark", "Warning", "Danger"],
        time_ranges: [
          "All Time",
          "Last Day",
          "Last Week",
          "Last Month",
          "Last Year"
        ],
        orders: [
          "A - B - C",
          "A - C - B",
          "B - A - C",
          "B - C - A",
          "C - A - B",
          "C - B - A"
        ],
        tweet_numbers: ["90", "80", "60", "40", "20", "5"]
      };
    },
    // Load variables values from the localStorage
    mounted() {
      if (localStorage.pallete) {
        globalState.pallete = localStorage.pallete;
      }
      if (localStorage.time_range) {
        globalState.time_range = localStorage.time_range;
      }
      if (localStorage.order) {
        globalState.order = localStorage.order;
      }
      if (localStorage.tweet_number) {
        globalState.tweet_number = localStorage.tweet_number;
      }
    }
  };
</script>

