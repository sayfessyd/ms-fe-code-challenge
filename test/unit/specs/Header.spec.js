import { mount } from "@vue/test-utils";
import Header from '@/components/Header.vue'

describe('Header.vue', () => {
  it('renders correctly', () => {
    const comp = mount(Header)
    expect(comp.html()).toMatchSnapshot();
  })
  it('renders correctly after changing the title state', () => {
    const comp = mount(Header)
    comp.setData({ title: "Hello!" });
    expect(comp.html()).toMatchSnapshot();
  })
})

