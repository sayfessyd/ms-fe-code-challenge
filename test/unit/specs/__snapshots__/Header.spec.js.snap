// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue renders correctly 1`] = `
<section class="hero is-small is-bold is-light">
    <div class="hero-body">
        <div class="container has-text-centered"><img src="../assets/Twitter_bird_icon.png" width="100px" alt="Twitter Logo">
            <h1 class="title">Twitter API</h1>
            <h2 class="subtitle">Last 30 tweets from @MakeSchool, @newsycombinator and @ycombinator</h2>
        </div>
    </div>
    <div class="hero-foot">
        <nav class="tabs is-boxed is-fullwidth">
            <div class="container has-text-centered">
                <ul class="is-responsive">
                    <li class="is-active">
                        <div class="control has-icons-left">
                            <div class="select is-small is-rounded"><select>
                                    <option>A - B - C</option>
                                    <option>A - C - B</option>
                                    <option>B - A - C</option>
                                    <option>B - C - A</option>
                                    <option>C - A - B</option>
                                    <option>C - B - A</option>
                                </select></div> <span class="icon is-small is-left"><i class="fas fa-sort-alpha-up"></i></span>
                        </div>
                    </li>
                    <li>
                        <div class="control has-icons-left">
                            <div class="select is-small is-rounded"><select>
                                    <option>All Time</option>
                                    <option>Last Day</option>
                                    <option>Last Week</option>
                                    <option>Last Month</option>
                                    <option>Last Year</option>
                                </select></div> <span class="icon is-small is-left"><i class="fas fa-clock"></i></span>
                        </div>
                    </li>
                    <li>
                        <div class="control has-icons-left">
                            <div class="select is-small is-rounded"><select>
                                    <option>90</option>
                                    <option>80</option>
                                    <option>60</option>
                                    <option>40</option>
                                    <option>20</option>
                                    <option>5</option>
                                </select></div> <span class="icon is-small is-left"><i class="fas fa-file"></i></span>
                        </div>
                    </li>
                    <li>
                        <div class="control has-icons-left">
                            <div class="select is-small is-rounded"><select>
                                    <option>Primary</option>
                                    <option>Info</option>
                                    <option>Light</option>
                                    <option>Dark</option>
                                    <option>Warning</option>
                                    <option>Danger</option>
                                </select></div> <span class="icon is-small is-left"><i class="fas fa-palette"></i></span>
                        </div>
                    </li>
                </ul>
            </div>
        </nav>
    </div>
</section>
`;

exports[`Header.vue renders correctly after changing the title state 1`] = `
<section class="hero is-small is-bold is-light">
    <div class="hero-body">
        <div class="container has-text-centered"><img src="../assets/Twitter_bird_icon.png" width="100px" alt="Twitter Logo">
            <h1 class="title">Hello!</h1>
            <h2 class="subtitle">Last 30 tweets from @MakeSchool, @newsycombinator and @ycombinator</h2>
        </div>
    </div>
    <div class="hero-foot">
        <nav class="tabs is-boxed is-fullwidth">
            <div class="container has-text-centered">
                <ul class="is-responsive">
                    <li class="is-active">
                        <div class="control has-icons-left">
                            <div class="select is-small is-rounded"><select>
                                    <option>A - B - C</option>
                                    <option>A - C - B</option>
                                    <option>B - A - C</option>
                                    <option>B - C - A</option>
                                    <option>C - A - B</option>
                                    <option>C - B - A</option>
                                </select></div> <span class="icon is-small is-left"><i class="fas fa-sort-alpha-up"></i></span>
                        </div>
                    </li>
                    <li>
                        <div class="control has-icons-left">
                            <div class="select is-small is-rounded"><select>
                                    <option>All Time</option>
                                    <option>Last Day</option>
                                    <option>Last Week</option>
                                    <option>Last Month</option>
                                    <option>Last Year</option>
                                </select></div> <span class="icon is-small is-left"><i class="fas fa-clock"></i></span>
                        </div>
                    </li>
                    <li>
                        <div class="control has-icons-left">
                            <div class="select is-small is-rounded"><select>
                                    <option>90</option>
                                    <option>80</option>
                                    <option>60</option>
                                    <option>40</option>
                                    <option>20</option>
                                    <option>5</option>
                                </select></div> <span class="icon is-small is-left"><i class="fas fa-file"></i></span>
                        </div>
                    </li>
                    <li>
                        <div class="control has-icons-left">
                            <div class="select is-small is-rounded"><select>
                                    <option>Primary</option>
                                    <option>Info</option>
                                    <option>Light</option>
                                    <option>Dark</option>
                                    <option>Warning</option>
                                    <option>Danger</option>
                                </select></div> <span class="icon is-small is-left"><i class="fas fa-palette"></i></span>
                        </div>
                    </li>
                </ul>
            </div>
        </nav>
    </div>
</section>
`;
